lib/kennel.rb

```diff
@@ -54,48 +54,88 @@ class << self
   self.err = $stderr
 
   class Engine
-    attr_accessor :strict_imports
+    def initialize(
+      update_generated: true,
+      show_plan: false,
+      require_confirm: nil,
+      update_datadog: false,
+      strict_imports: true
+    )
+      @update_generated = update_generated
+      @show_plan = show_plan
+      @require_confirm =
+        if require_confirm.nil?
+          default_require_confirm?
+        else
+          require_confirm
+        end
+      @update_datadog = update_datadog
+      @strict_imports = strict_imports
+    end
+
+    def run
+      if !update_generated? && !show_plan? && !update_datadog?
```

Comment:
should fail and say "you did not ask for anything" ?

Reply:
It could, that's true.

Or it could do nothing (e.g. just `return nil`).

Or it could do something useful (validate that the parts can be built and are valid). Which is what it does.